packages/cli/src/utils/telemetry.ts

```diff
@@ -0,0 +1,126 @@
+import { ux } from '@oclif/core'
+import chalk from 'chalk'
+import fs from 'fs'
+import path from 'path'
+import packageJson from '../../package.json'
+import { readConfig } from './config'
+
+const debug = require('debug')('celocli:telemetry')
+
+export type TelemetryOptions = {
+  command: string
+  success: boolean
+  version: string
+}
+
+const getTelemetryOptions = (command: string, success: boolean): TelemetryOptions => {
+  return {
+    command,
+    success,
+    version: packageJson.version,
+  }
+}
+
+const TELEMETRY_TIMEOUT = 1000 // 1 second
+const TELEMETRY_PRINTED_FILE = '.telemetry'
+const TELEMETRY_URL = 'https://pag.mainnet.celo-testnet.org/metrics'
+
+const UNKNOWN_COMMAND = '_unknown'
+
+const telemetryInformationAlreadyPrinted = (configDir: string) => {
+  try {
+    return fs.existsSync(path.join(configDir, TELEMETRY_PRINTED_FILE))
+  } catch (err) {
+    debug(`Failed to check if .telemetry file exists: ${err}`)
+  }
+
+  return false
+}
+
+const markTelemetryInformationAsPrinted = (configDir: string) => {
+  try {
+    fs.writeFileSync(path.join(configDir, TELEMETRY_PRINTED_FILE), '')
+  } catch (err) {
+    debug(`Failed to write .telemetry file: ${err}`)
+  }
+}
+
+export const reportUsageStatisticsIfTelemetryEnabled = (
+  configDir: string,
+  success: boolean,
+  command: string = UNKNOWN_COMMAND
+) => {
+  if (process.env.TELEMETRY_ENABLED === '0') {
+    return
+  }
+
+  const config = readConfig(configDir)
+
+  if (config.telemetry === true) {
+    const telemetry = getTelemetryOptions(command, success)
+
+    // Only show the information upon first usage
+    if (!telemetryInformationAlreadyPrinted(configDir)) {
+      printTelemetryInformation()
+      markTelemetryInformationAsPrinted(configDir)
+    }
+
+    const telemetryData = `celocli_invocation{success="${
+      telemetry.success ? 'true' : 'false'
+    }", version="${telemetry.version}", command="${telemetry.command}"} 1`
+
+    debug(`Sending telemetry data: ${telemetryData}`)
+
+    const controller = new AbortController()
+    const timeout = setTimeout(() => {
+      controller.abort()
+    }, TELEMETRY_TIMEOUT)
+
+    return fetch(process.env.TELEMETRY_URL ?? TELEMETRY_URL, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/octet-stream',
+      },
+      signal: controller.signal,
+      body: `
+${telemetryData}
```

Comment:
Im surprised you can send plain text as `octet-stream`, thought you had to transmit as a blob/buffer kind of body